processing/src/main/java/org/apache/druid/query/context/ResponseContext.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.context;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.druid.java.util.common.jackson.JacksonUtils;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * The context for storing and passing data between chains of {@link org.apache.druid.query.QueryRunner}s.
+ * The context is also transferred between Druid nodes with all the data it contains.
+ * All the keys associated with data inside the context should be stored here.
+ */
+public abstract class ResponseContext
+{
+  protected abstract Map<String, Object> getDelegate();
+
+  public static final String CTX_UNCOVERED_INTERVALS = "uncoveredIntervals";
```

Comment:
Have you considered creating an enum and allowing to put only enum keys?

Reply:
Yes, thought about that. But then found out that a lot of tests use ResponseContext with their own dummy string keys and decided to make as fewer changes as possible. This update might be included in a new PR along with integrating ResponseContext into QueryPlus

Reply:
Please add a comment explaining this and mentioning that an enum may be considered in the future.

Reply:
Added the comment